src/js/components/sharedComponents/stickyHeader/ShareIcon.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { Picker } from "data-transparency-ui";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { debounce } from "lodash";
+
+import { socialShareOptions, getSocialShareFn, getBaseUrl } from 'helpers/socialShare';
+
+const propTypes = {
+    slug: PropTypes.string,
+    email: PropTypes.shape({
+        subject: PropTypes.string,
+        body: PropTypes.string
+    })
+};
+
+const ShareIcon = ({
+    slug,
+    email: { subject, body }
+}) => {
+    const [showConfirmationText, setConfirmationText] = useState(false);
+    const hideConfirmationText = debounce(() => setConfirmationText(false), 1750);
+
+    useEffect(() => {
+        if (showConfirmationText) {
+            hideConfirmationText();
+        }
+        return hideConfirmationText.cancel;
+    }, [showConfirmationText, setConfirmationText, hideConfirmationText]);
+
+    const getCopyFn = () => {
+        document.getElementById('slug').select();
+        document.execCommand("copy");
+        setConfirmationText(true);
+    };
+
+    const socialSharePickerOptions = socialShareOptions.map((option) => {
+        if (option.name === 'copy') {
+            return {
+                ...option,
+                onClick: getCopyFn
+            };
+        }
+        if (option.name === 'email') {
+            const onClick = getSocialShareFn(option.name).bind(null, {
+                subject,
+                body
+            });
+            return {
+                ...option,
+                onClick
+            };
+        }
+        return {
+            ...option,
+            onClick: getSocialShareFn(option.name).bind(null, slug)
+        };
+    });
+
+    return (
+        <div className="sticky-header__toolbar-item">
+            <input id="slug" type="text" className="text" style={{ position: 'absolute', right: '9999px', opacity: 0 }} value={getBaseUrl(slug)} />
```

Comment:
This is throwing a console warning since there's no `onChange` provided. Looks like we should be able to set it to `readOnly` to get rid of the warning.